Remora.Rest/Json/Internal/BoundDataObjectConverterOptions.cs

```diff
@@ -0,0 +1,41 @@
+//
+//  BoundDataObjectConverterOptions.cs
+//
+//  Author:
+//       Jarl Gullberg <jarl.gullberg@gmail.com>
+//
+//  Copyright (c) Jarl Gullberg
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using Remora.Rest.Json.Reflection;
+
+namespace Remora.Rest.Json.Internal;
+
+/// <summary>
+/// Encapsulates all data needed to construct a <see cref="BoundDataObjectConverter{TInterface, TImplementation}"/>.
+/// </summary>
+/// <typeparam name="TInterface">The interface that is seen in the objects for the converter.</typeparam>
+/// <param name="DTOFactory">The DTO factory.</param>
+/// <param name="AllowExtraProperties">Whether extra undefined properties should be allowed.</param>
+/// <param name="WriteProperties">Properties relevant when writing the DTO to JSON.</param>
+/// <param name="ReadProperties">Properties relevant when reading the DTO from JSON.</param>
+internal record BoundDataObjectConverterOptions<TInterface>
```

Comment:
What's the use for this record, if it's only going to be deconstructed again in an internal class's constructor? I think it's more appropriate to just have straight arguments.

Reply:
I prefer not to have a lot of arguments to methods, but I'll change it.

Reply:
I get that, but the method still has a lot of arguments; they're just hidden by a level of indirection.

Reply:
Yeah, that's fair.

Reply:
Arguments passed directly now:
https://github.com/Remora/Remora.Rest/blob/916ee44822b0206e371039ab37534decdacdb646/Remora.Rest/Json/DataObjectConverter.cs#L536-L542